### PR TITLE
add pairwise judgement environment

### PR DESCRIPTION
**Pull request by teknium1**
## PR Type

- [x] RL Environment PR - Complete Environment Snapshot & Zero-Training sections
- [ ] Non-Environment PR - Complete Description, Related Issues & Type of Change sections

---

### Description
Implements a new pairwise judgment environment for training models to evaluate and compare AI assistant responses. The environment presents models with a user question and four AI responses (A, B, C, D), asking them to judge which response is best. Models must provide their judgment in the format `[[A]]`, `[[B]]`, `[[C]]`, or `[[D]]` with optional reasoning in `<think></think>` tags.

Adapted from the official rewardbench repo here: https://github.com/allenai/reward-bench

---

## 🔖 Environment Snapshot

| Field | Your Entry |
|-------|------------|
| **Environment Name** | `pairwise_judgement` |
| **Short Description** | Trains models to judge which of 4 AI assistant responses best answers a user question. |
| **Category** | RLAIF |
| **Dataset Needed?** | Yes - `allenai/reward-bench-2` (Apache 2.0 license) |
| **External Deps** | `datasets`, `wandb`, `tqdm` |
| **Environmental Variables** | None |
| **Compute Footprint Estimate** | <2 GB RAM, standard text processing workload |

**Citation**: 
```bibtex
@misc{lambert2024rewardbenchevaluatingrewardmodels,
      title={RewardBench: Evaluating Reward Models for Language Modeling}, 
      author={Nathan Lambert and Valentina Pyatkin and Jacob Morrison and LJ Miranda and Bill Yuchen Lin and Khyathi Chandu and Nouha Dziri and Sachin Kumar and Tom Zick and Yejin Choi and Noah A. Smith and Hannaneh Hajishirzi},
      year={2024},
      eprint={2403.13787},
      archivePrefix={arXiv},
      primaryClass={cs.LG},
      url={https://arxiv.org/abs/2403.13787}, 
}
```

## 🧪 Zero-Training Test Results

<details>

**W&B Link:** 
*To be provided after running evaluation*

**Examples of the Environment scoring a good example and a bad example:**

**Good Example (Score: 1.0):**
- **Question**: "What is the capital of France?"
- **Responses**: A: "Paris", B: "London", C: "Berlin", D: "Madrid"  
- **Ground Truth**: A
- **Model Output**: "Looking at this question, the capital of France is clearly Paris. [[A]]"
- **Predicted**: A
- **Score**: 1.0 ✅

**Bad Example (Score: 0.0):**
- **Question**: "How do you fix a memory leak in Python?"
- **Responses**: A: "Use profilers and proper cleanup", B: "Restart computer", C: "Impossible in Python", D: "Reinstall Python"
- **Ground Truth**: A  
- **Model Output**: "Python handles memory automatically, so [[C]]"
- **Predicted**: C
- **Score**: 0.0 ❌

**Format Error Example (Score: 0.0):**
- **Model Output**: "I think A is the best response because it's comprehensive"
- **Predicted**: error (no [[letter]] format)
- **Score**: 0.0 ❌

</details>

---

## ✅ Developer & Reviewer Checklist

- [x] Code follows project style (black, isort, flake8 pass with pre-commit)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Docstrings added for all new public classes / functions
- [x] If .env vars required, did you add it to the .env.example in repo root?

**Additional Environment-Specific Checklist:**
- [x] Environment implements all required base class methods (`setup`, `get_next_item`, `collect_trajectories`, `score`, `evaluate`)
- [x] Sample-level evaluation logging implemented with `evaluate_log` method
- [x] Comprehensive WandB metrics tracking implemented
- [x] Thinking mode support with proper `<think></think>` tag parsing
- [x] Robust error handling for format compliance (models must use `[[A]]`, `[[B]]`, `[[C]]`, `[[D]]` format)
- [x] Synthetic training data generation with varied examples
- [x] Reward-bench-2 dataset integration for evaluation
- [x] Proper shuffling and randomization of response order to avoid position bias
- [x] Support for both direct and thinking modes via configuration

### CODE REVIEW

**Review by hjc-puro — APPROVED** [review]
LGTM
